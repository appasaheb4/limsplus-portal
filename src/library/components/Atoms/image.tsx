import React from "react";

interface ImageProps {
  source: string;
  fit: "cover" | "contain" | "auto";
  position:
    | "bottom"
    | "center"
    | "left"
    | "left-bottom"
    | "left-top"
    | "right"
    | "right-bottom"
    | "right-top"
    | "top";
  background?: "fill" | "plain";
  className?: string
}

const Image = (props: ImageProps) => {
  const backgroundColorClass =
    props.background === "fill" ? "bg-gray-200" : "bg-transparent";
  const backgroundPositionClass = `bg-${props.position}`;
  const backgroundSizeClass =
    props.fit === "cover"
      ? "bg-cover"
      : props.fit === "contain"
      ? "bg-contain"
      : "";

  return (
    <div
      style={{ backgroundImage: `url(${props.source})` }}
      className={`${props.className} bg-no-repeat ${backgroundSizeClass} ${backgroundColorClass} ${backgroundPositionClass}`}
    ></div>
  );
};

export default Image;
