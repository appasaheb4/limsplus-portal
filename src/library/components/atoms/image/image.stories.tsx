import React from 'react';
import {ComponentStory, ComponentMeta} from '@storybook/react';
import {Image, ImageBackground} from './image.component';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Atoms/Image',
  component: Image,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: {control: 'color'},
  },
} as ComponentMeta<typeof Image>;

export const _Image: ComponentStory<typeof Image> = () => (
  <Image
    widht={200}
    height={200}
    source='https://cdn.pixabay.com/photo/2019/03/28/22/23/link-4088190_1280.png'
    onClick={() => console.log('Image')}
  />
);

export const _ImageBackground: ComponentStory<typeof Image> = () => (
  <ImageBackground
    position='center'
    fit='cover'
    source='https://images.unsplash.com/photo-1553095066-5014bc7b7f2d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8d2FsbCUyMGJhY2tncm91bmR8ZW58MHx8MHx8&w=1000&q=80'
  />
);
