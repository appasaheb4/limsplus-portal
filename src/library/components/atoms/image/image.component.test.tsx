import React from 'react';
import {render} from '@utils';
import {Image, ImageBackground} from './image.component';

it('render image correctly', () => {
  const image = render(<Image source="" onClick={() => jest.fn()} />);
  expect(image).toMatchSnapshot();
});
it('render imageBackground correctly', () => {
  const imageBackground = render(
    <ImageBackground source="" position="center" fit="cover" />,
  );
  expect(imageBackground).toMatchSnapshot();
});
