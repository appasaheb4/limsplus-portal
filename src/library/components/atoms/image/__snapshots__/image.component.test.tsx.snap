// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Background component render imagebackground correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="undefined bg-no-repeat  bg-transparent bg-bottom"
        style="background-image: url(nowec);"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="undefined bg-no-repeat  bg-transparent bg-bottom"
      style="background-image: url(nowec);"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Image component render image correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="limsplus"
        class="m-4 rounded-md object-fill"
        src="csdfrf"
        style="display: block; position: relative; left: 10%;"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="limsplus"
      class="m-4 rounded-md object-fill"
      src="csdfrf"
      style="display: block; position: relative; left: 10%;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
